File: code/cpp/ch18/practice/warmup_01_assign_cookies.cpp
/*
 * Warmup 1: Assign Cookies
 * Chapter 18: Greedy Algorithms — The Smart Shortcut
 *
 * PROBLEM: Maximize content children. Cookie >= greed means content.
 * EXAMPLES: solve({1,2,3}, {1,1}) -> 1
 * INSTRUCTIONS: Replace the body of solve() with your solution.
 */

#include <algorithm>
#include <iostream>
#include <vector>
using namespace std;

int solve(vector<int> greed, vector<int> cookies) {
    // TODO: Replace this with your solution
    return 0;
}

// -- Do not change anything below this line --------------------------
int main() {
    int n, m;
    cin >> n;
    vector<int> greed(n);
    for (int i = 0; i < n; i++) cin >> greed[i];
    cin >> m;
    vector<int> cookies(m);
    for (int i = 0; i < m; i++) cin >> cookies[i];
    cout << solve(greed, cookies) << endl;
    return 0;
}

File: code/python/ch18/practice/warmup_01_assign_cookies.py
"""
Warmup 1: Assign Cookies
==========================
Chapter 18: Greedy Algorithms — The Smart Shortcut

PROBLEM
-------
You have children with greed factors and cookies with sizes.
A child is content if cookie size >= greed factor.
Maximize content children.

EXAMPLES
--------
>>> solve([1, 2, 3], [1, 1])
1
>>> solve([1, 2], [1, 2, 3])
2

CONSTRAINTS
-----------
- 0 <= len(greed), len(cookies) <= 10^4
"""


def solve(greed: list[int], cookies: list[int]) -> int:
    """Return the maximum number of content children."""
    pass  # TODO: Replace this with your solution


# ── Do not change anything below this line ──────────────────────────
if __name__ == "__main__":
    greed = list(map(int, input().strip().split()))
    cookies = list(map(int, input().strip().split()))
    print(solve(greed, cookies))
